lecture_2/homework/config/application.rb

```diff
@@ -21,7 +23,10 @@ module Lecture1
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 5.2
-
+    config.autoload_paths += %W[#{Rails.root}/app/models/warriors]
```

Comment:
Widzę, że próbowałeś fajnie pogrupować modele w aplikacji i trafiłeś na opory ze strony railsów. Konwencja jest taka, że jeżeli umieszczamy klasę w folderze to powinna być w module. Czyli wewnątrz pliku `app/models/warriors/hussar.rb` będziemy mieli plik w którym znajdziemy coś takiego
```ruby
module Warriors
  class Hussar
  ...
```
Wtedy takiego wojownika tworzymy `Warriors::Hussar.new...`

Analogicznie, jeśli nasza klasa nazywa się po prostu `Warrior` to powinniśmy umieścić ją w `app/models/warrior.rb`.

Po zastosowaniu się do tych dwóch wytycznych będziemy mieć klasę bazową `Warrior` i wszystkie klasy dziedziczące w postaci `Warriors::Something`. Wtedy już nie trzeba dodawać ścieżek do autoloadu 😃